R/retrieve.R

```diff
@@ -28,12 +28,16 @@ retrieve <- function(packages) {
   }
   renv_scope_options(HTTPUserAgent = agent)
 
-  writef("Downloading %s package(s) and their dependencies", length(packages))
+  writef(header("Downloading package(s)"))
   # TODO: parallel?
   handler <- state$handler
   for (package in packages)
     handler(package, renv_retrieve_impl(package))
-  writef(c("Done", ""))
+
+  if (is.null(state$downloaded)) {
+    writef("Nothing needed")
```

Comment:
I think we also need to print something if no downloads actually occurred.

Reply:
```suggestion
    writef("[no downloads required]")
```

I like using adornments for placeholder text like this just to help make it clear that it's not "regular" output. I'm open to alternatives on wordsmithing though.